sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1531,6 +1531,102 @@ def _df_getitem_unicode_idx_impl(self, idx):
     ty_checker.raise_exc(idx, expected_types, 'idx')
 
 
+def df_getitem_tuple_iat_codegen(self, row, col):
+    """
+    Example of generated implementation:
+        def _df_getitem_tuple_iat_impl(self, idx):
+            row, _ = idx
+            data = get_dataframe_data(self._dataframe, 1)
+            res_data = pandas.Series(data)
+            return res_data.iat[row]
+    """
+    func_lines = ['def _df_getitem_tuple_iat_impl(self, idx):']
+    func_lines += ['  row, _ = idx']
+    if self.columns:
+        func_lines += [
+            f"  data = get_dataframe_data(self._dataframe, {col})",
+            f"  res_data = pandas.Series(data)",
+            f"  return res_data.iat[row]",
+        ]
+
+    func_text = '\n'.join(func_lines)
+
+    global_vars = {'pandas': pandas,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+def gen_df_getitem_iat_impl(self, row, col, func_name):
+    func_text, global_vars = df_getitem_tuple_iat_codegen(self, row, col)
+    loc_vars = {}
+    exec(func_text, global_vars, loc_vars)
+    _reduce_impl = loc_vars[func_name]
+
+    return _reduce_impl
+
+
+@sdc_overload(operator.getitem)
+def sdc_pandas_dataframe_accessor_getitem(self, idx):
+    if not isinstance(self, DataFrameGetitemAccessorType):
+        return None
+
+    accessor = self.accessor.literal_value
+
+    if accessor == 'iat':
+        if isinstance(idx, types.Tuple):
+            row = idx[0].literal_value
+            col = idx[1].literal_value
+            return gen_df_getitem_iat_impl(self.dataframe, row, col, '_df_getitem_tuple_iat_impl')
```

Comment:
Let's hide `'_df_getitem_tuple_iat_impl'` somewhere inside of the implementation generator.